fields/types/list/ListType.js

```diff
@@ -0,0 +1,225 @@
+		if (options.type.name === 'html' && options.wysiwyg) {
+			field.list.fieldTypes.wysiwyg = true;
+		}
+		// Tell the Field that it is nested, this changes the constructor slightly
+		options._isNested = true;
+		options._nestedSchema = itemSchema;
+		return new options.type(field.list, path, options);
+	}
+
+	Object.keys(fieldsSpec).forEach(function (path) {
+		if (!fieldsSpec[path]) {
+			throw new Error(
+				'Invalid value for nested schema path `' + path + '` in `'
+				+ field.list.key + '.' + field.path + '`.\n'
+				+ 'Did you misspell the field type?\n'
+			);
+		}
+		if (isReserved(path)) {
+			throw new Error(
+				'Nested schema path ' + path + ' on field '
+				+ field.list.key + '.' + field.path + ' is a reserved path'
+			);
+		}
+		var newField = createField(path, fieldsSpec[path]);
+		fields[path] = newField;
+		fieldsArray.push(newField);
+	});
+
+	if (this.options.decorateSchema) {
+		this.options.decorateSchema(itemSchema);
+	}
+
+	schema.add(this._path.addTo({}, [itemSchema]));
+	this.bindUnderscoreMethods();
+};
+
+/**
+ * Provides additional properties for the Admin UI
+ */
+list.prototype.getProperties = function (item, separator) {
+	var fields = {};
+	this.fieldsArray.forEach(function (field) {
+		fields[field.path] = field.getOptions();
+	});
+	return {
+		fields: fields,
+	};
+};
+
+/**
+ * Formats the field value
+ */
+list.prototype.format = function (item, separator) {
+	// TODO: How should we format nested items? Returning length for now.
+	var items = item.get(this.path) || [];
+	return utils.plural(items.length, '* Value', '* Values');
+};
+
+// TODO: How should we filter list values?
+/*
+list.prototype.addFilterToQuery = function (filter) { };
+*/
+
+/**
+ * Asynchronously confirms that the provided value is valid
+ */
+list.prototype.validateInput = function (data, callback) {
+	// TODO
+	// var value = this.getValueFromData(data);
+	var result = true;
+	utils.defer(callback, result);
+};
+
+/**
+ * Asynchronously confirms that the a value is present
+ */
+list.prototype.validateRequiredInput = function (item, data, callback) {
+	// TODO
+	// var value = this.getValueFromData(data);
+	var result = true;
+	utils.defer(callback, result);
+};
+
+list.prototype.getData = function (item) {
+	var items = item.get(this.path);
+	var fieldsArray = this.fieldsArray;
+	return items.map(function (i) {
+		var result = { id: i.id };
+		for (var field of fieldsArray) {
+			result[field.path] = field.getData(i);
+		}
+		return result;
+	});
+};
+
+/**
+ * Updates the value for this field in the item from a data object.
+ * If the data object does not contain the value, then the value is set to empty array.
+ */
+list.prototype.updateItem = function (item, data, files, callback) {
+	if (typeof files === 'function') {
+		callback = files;
+		files = {};
+	}
+
+	var field = this;
+	var values = this.getValueFromData(data);
+	// Don't update the value when it is undefined
+	if (values === undefined) {
```

Comment:
This means you can't remove the last item in a list. All the (Text|Number|Date)Array types fix this by treating `values === undefined` the same as `=== null`.